File: easy/704-binary-search/704-binary-search.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer}
def search(nums, target)
    l = 0
    r = nums.size - 1
    while l <= r do 
        m = (l + r) / 2
        return m if nums[m] == target
        nums[m] < target ? l = m + 1 : r = m - 1
    end
    -1
end

File: easy/704-binary-search/704-binary-search.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var search = function(nums, target) {
    for (let [l,r] = [0,nums.length-1]; l <= r;) {
        const m = (l + r) >> 1;
        if (nums[m] === target) return m;
        nums[m] < target ? l = m + 1 : r = m - 1;
    }
    return -1;
};